tests/modules/RouterTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ *
+ * RouterTest.php is unit test file for Router module (modules/Router.php).
+ *
+ */
+namespace Tests\FirstAide;
+
+require_once str_replace('tests/modules', '', dirname(__FILE__)).'/includes/constants.php';
+
+/**
+ * Importing required modules/wrappers
+ */
+use FirstAide\User;
+use FirstAide\Router;
+use FirstAide\MysqlDatabase;
+use PHPUnit\Framework\TestCase;
+use Tests\MysqlWrapperMockTrait;
+
+class RouterTest extends TestCase
+{
+    /**
+     * Importing mock MySQL wrapper
+     */
+    use MysqlWrapperMockTrait;
+
+    /**
+     * @dataProvider mysqlMockProvider
+     */
+    public function testGetPage($databaseMock)
```

Comment:
The test case covers the index path, but when we look at the testing coverage, there is a lot of branching and routing that is untested and might lead to bugs. Unit tests should catch those issues, but, unit tests overall are powerful only when there is good code coverage. 

As you can see, each path returns different results based on the query/path. It might not be a good idea to write test cases for each path as it would be redundant. Instead, it might be a good idea to test the router functionality separately from the model creation. 

Ideally we will need to start with separating out the model creation logic from the Router.php file so that each component can be tested individually? This would help in reducing redundant test cases. We can add a task in backlog for this. 

For now, here are the changes needed:
1. Code needs to be modular. And Each module needs to be short and not too big like "getPage" method. To achieve this, each page creation (that comes under a case) should be put in a method on its own. 

For example, the case **self::HOME** should call a method called **getHomePage** that would return the data for this page. 

2. Once the modularity is achieved, unit tests need to be written for methods that return different sets of data. For example, case case self::PAGE_SEEKING_STAFF_SUPPORT and self::PAGE_SEXUAL_HARASSMENT return same set of data, so one unit test covering either of the methods for these two pages should be good. But the case self::PAGE_CIRCLE_OF_TRUST and other cases that return a different set of data should have another unit test and so on.